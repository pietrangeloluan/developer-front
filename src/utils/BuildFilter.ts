export default function (params: any) {
  let filter = {}
  for (const key in params) {
    if (Object.hasOwnProperty.call(params, key)) {
      const parameter = params[key]
      if (parameter) {
        filter = { ...filter, [key]: parameter }
      }
    }
  }
  return filter
}
