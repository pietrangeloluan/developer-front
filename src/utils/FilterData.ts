export default function (obj, filter) {
  const filtered = obj.filter(item => {
    for (const key in filter) {
      if (item[key] === undefined || !filter[key].includes(item[key])) {
        return false
      }
    }
    return true
  })

  return filtered
}
