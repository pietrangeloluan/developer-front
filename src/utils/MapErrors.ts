export function mapErrors(errors, context) {
  const errorKeys = Object.keys(errors)
  const regex = new RegExp(`${context}\\[\\d+\\]`, 'g')
  const contextErrors = errorKeys.filter(err => err.match(regex))

  if (!contextErrors.length) return [{}]

  const updated = []

  contextErrors.forEach(err => {
    const found = [...err.matchAll(/\[(\d+)\]\.(\w+)/g)]

    if (found.length) {
      found[0].shift()
      const errIndex = found[0][0]
      const errField = found[0][1]

      updated[errIndex] = {
        ...updated[errIndex],
        [errField]: errors[`${context}[${errIndex}].${errField}`]
      }
    }
  })

  return updated
}
