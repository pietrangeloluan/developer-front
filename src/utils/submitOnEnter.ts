import { KeyboardEvent } from 'react'

export function submitOnEnter(fn: Function, args: any[] = []) {
  return function (e: KeyboardEvent) {
    if (e.key === 'Enter') fn.apply(args)
  }
}
