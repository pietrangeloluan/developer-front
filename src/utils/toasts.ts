import { toast } from 'react-toastify'

export const toasts = {
  created() {
    return toast.success('Dados cadastrados com sucesso.', {
      autoClose: 5 * 1000
    })
  },
  deleted() {
    return toast.success('Dados exclu√≠dos com sucesso.', {
      autoClose: 5 * 1000
    })
  },
  edited() {
    return toast.success('Dados editados com sucesso.', { autoClose: 5 * 1000 })
  },
  generalFail({ code = 'UNKNOWN_ERROR' } = {}) {
    return toast.error(
      'Ocorreu uma falha inesperada. Por favor, aguarde alguns minutos de tente novamente.',
      { autoClose: 5 * 1000, toastId: code }
    )
  }
}
