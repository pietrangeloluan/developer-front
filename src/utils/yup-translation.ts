/* eslint-disable no-template-curly-in-string */
import { setLocale } from 'yup'

const translation = {
  array: {
    max: '${path} deve conter no máximo ${max} itens',
    min: '${path} deve conter no mínimo ${min} itens'
  },
  date: {
    max: '${path} deve ser anterior à ${max}',
    min: '${path} deve ser mais recente que ${min}',
    required: '${path} é obrigatório'
  },
  mixed: {
    default: '${path} é inválido',
    notOneOf: '${path} não pode ser igual à um destes valores: ${values}',
    oneOf: '${path} deve ser igual à um dos seguintes valores: ${values}',
    required: '${path} é obrigatório'
  },
  number: {
    integer: '${path} deve ser inteiro',
    lessThan: '${path} deve ser menor à ${less}',
    max: '${path} deve ser no máximo ${max}',
    min: '${path} deve ser no mínimo ${min}',
    moreThan: '${path} deve ser maior à ${more}',
    negative: '${path} deve ser negativo',
    notEqual: '${path} não pode ser igual à ${notEqual}',
    positive: '${path} deve ser positivo',
    required: '${path} é obrigatório'
  },
  string: {
    email: '${path} não contém um e-mail válido',
    length: '${path} deve conter exatamente ${length} caracteres',
    lowercase: '${path} deve ser maiúsculo',
    max: '${path} deve conter no máximo ${max} caracteres',
    min: '${path} deve conter pelo menos ${min} caracteres',
    trim: '${path} não deve conter espaços em branco em início ou final.',
    uppercase: '${path} deve ser minúsculo',
    url: '${path} não contém uma URL válida'
  }
}

setLocale(translation)
