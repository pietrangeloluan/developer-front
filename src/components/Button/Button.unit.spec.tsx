import { fireEvent, screen } from '@testing-library/react'
import faker from 'faker'

import { renderWithThemeProvider } from '__tests__/utils'

import { Button } from '.'

describe('Button', () => {
  it('should render the component', () => {
    const children = faker.lorem.words()
    renderWithThemeProvider(Button, { children })

    const found = screen.getByRole('button', { name: children })
    expect(found).toBeInTheDocument()
  })

  it('should match the snapshot', () => {
    const { container } = renderWithThemeProvider(Button, {
      children: 'Hello World!'
    })
    expect(container).toMatchSnapshot()
  })

  it('should has blue variant as default', () => {
    const { container } = renderWithThemeProvider(Button, {
      children: 'Hello World!'
    })
    expect(container.firstChild).toHaveClass('blue')
  })

  it('should has another variant when receives it', () => {
    const variant = 'gray'
    const { container } = renderWithThemeProvider(Button, {
      children: faker.lorem.words(),
      variant
    })
    expect(container.firstChild).toHaveClass(variant)
  })

  it('should has medium size as default', () => {
    const { container } = renderWithThemeProvider(Button, {
      children: 'Hello World!'
    })
    expect(container.firstChild).toHaveClass('medium')
  })

  it('should has another size when receives it', () => {
    const size = 'large'
    const { container } = renderWithThemeProvider(Button, {
      children: faker.lorem.words(),
      size
    })
    expect(container.firstChild).toHaveClass(size)
  })

  it('should call onClick when get clicked', () => {
    // arrange
    // arrumando todas as 'dependÃªncias' do teste
    const stub = jest.fn()
    const { container } = renderWithThemeProvider(Button, {
      children: faker.lorem.words(),
      onClick: stub
    })

    // act
    // executando o que precisa ser testado
    fireEvent.click(container.firstChild)

    // assert
    // verificando o resultado
    expect(stub).toHaveBeenCalledTimes(1)
  })
})
