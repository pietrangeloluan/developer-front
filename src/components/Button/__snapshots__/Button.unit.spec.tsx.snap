// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
.c0 {
  outline: none;
  border-radius: 0.5rem;
  border: none;
  font-size: 1.4rem;
  font-weight: 600;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition: background-color 400ms,color 400ms,-webkit-transform 400ms;
  -webkit-transition: background-color 400ms,color 400ms,transform 400ms;
  transition: background-color 400ms,color 400ms,transform 400ms;
  will-change: background-color,color,transform;
  margin: 0.8rem;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c0.blue {
  background-color: #3684ab;
  color: white;
}

.c0.blue:hover {
  background-color: #00577c;
}

.c0.green {
  background-color: #3fab36;
  color: white;
}

.c0.green:hover {
  background-color: #007b00;
}

.c0.gray {
  background-color: #707070;
  color: white;
}

.c0.gray:hover {
  background-color: #454545;
}

.c0:active {
  -webkit-transform: scale(0.88);
  -ms-transform: scale(0.88);
  transform: scale(0.88);
}

.c0.large {
  padding: 1.4rem 10rem;
  font-size: 1.6rem;
}

.c0.medium {
  padding: 1.1rem 3.2rem;
}

.c0.small {
  padding: 0.7rem 2.8rem;
}

.c0.xsmall {
  box-shadow: 0px 3px 6px #c1c5c4;
  font-weight: 400;
  padding: 0.1rem 1.1rem;
}

<div>
  <button
    class="c0 blue medium"
  >
    Hello World!
  </button>
</div>
`;
